components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupFacetImpl.java

```diff
@@ -80,10 +95,26 @@ protected void doValidate(final Configuration configuration) throws Exception {
   @Override
   protected void doConfigure(final Configuration configuration) throws Exception {
     config = facet(ConfigurationFacet.class).readSection(configuration, CONFIG_KEY, Config.class);
+
+    cacheController = new CacheController(config.contentMaxAge, null);
+
     log.debug("Config: {}", config);
   }
 
   @Override
+  protected void doUpdate(final Configuration configuration) throws Exception {
+    // detect member changes
+    super.doUpdate(configuration);
+
+    // form hash out of member names: this will change on member change and on member reordering too
+    final String cacheToken = SHA1.function().hashString(
+        String.valueOf(config.memberNames),
+        Charsets.UTF_8
+    ).toString();
+    cacheController.mayInvalidateCache(cacheToken);
```

Comment:
With this change I think it makes sense to move the generation of `cacheToken` into `doConfigure` where the `CacheController` constructor can consume it. Since `doConfigure` is called on both initialisation and update we wouldn't need to override `doUpdate` and this would also remove the need for the `mayInvalidateCache` method.